### PR TITLE
feat: use uppercase acronyms in template functions

### LINKED ISSUE #84: misc: fix methods cases
### Please exprime yourself

Right now, it's

- toJson
- toYaml
- fromJson
- ...

But these are acronyms

It should be
- toJSON
- toYAML
- fromJSON

The code is OK I think. My problem is the identifier of methods in the template engine

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ccoVeille**
## Description

See discussions in 
- #84 

## Changes

The functions are now using uppercase acronym, old ones are aliased, but marked as deprecated.

## Checklist
- [X] I have read the **CONTRIBUTING.md** document.
- [X] My code follows the code style of this project.
- [?] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have updated the documentation accordingly.
- [?] This change requires a change to the documentation on the website.

### CODE REVIEW

**Comment by ccoVeille** [comment]
Please note, I didn't update some godoc that currently mentions the old syntax because you are working on updating them to url via

- #97

**Comment by ccoVeille** [comment]
I think you can review now the other PR is mergef

**Review by 42atomys — APPROVED** [review]
LGTM ✅ 

I just change the method signature to match the convention too 👀 and update doc from #97

File: docs/registries/encoding.md [hunk 1]
```diff
@@ -75,114 +75,114 @@ The function decodes a Base32 encoded string back to its original form. If the i
 {% endtab %}
 {% endtabs %}
 
-### <mark style="color:purple;">fromJson</mark>
+### <mark style="color:purple;">fromJSON</mark>
 
 The function converts a JSON string into a corresponding Go data structure, enabling easy manipulation of the JSON data in a Go environment.
 
-<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">FromJson(v string) (any, error)
+<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">fromJSON(v string) (any, error)
```
Comment (ccoVeille) [inline docs/registries/encoding.md:82]:
```suggestion
<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">FromJSON(v string) (any, error)
```

Maybe I was wrong here

File: docs/registries/encoding.md [hunk 2]
```diff
@@ -75,114 +75,114 @@ The function decodes a Base32 encoded string back to its original form. If the i
...
 {% endtab %}
 {% endtabs %}
 
-### <mark style="color:purple;">toPrettyJson</mark>
+### <mark style="color:purple;">toPrettyJSON</mark>
 
 The function converts a Go data structure into a pretty-printed JSON string, formatting the output with indentation and line breaks for better readability.
 
-<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">ToPrettyJson(v any) (string, error)
+<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">toPrettyJSON(v any) (string, error)
```
Comment (ccoVeille) [inline docs/registries/encoding.md:114]:
```suggestion
<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">ToPrettyJSON(v any) (string, error)
```

File: docs/registries/encoding.md [hunk 3]
```diff
@@ -75,114 +75,114 @@ The function decodes a Base32 encoded string back to its original form. If the i
...
 {% endtab %}
 {% endtabs %}
 
-### <mark style="color:purple;">toRawJson</mark>
+### <mark style="color:purple;">toRawJSON</mark>
 
 The function converts a Go data structure into a JSON string without escaping HTML characters, preserving the raw content as it is.
 
-<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">ToRawJson(v any) (string, error)
+<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">toRawJSON(v any) (string, error)
```
Comment (ccoVeille) [inline docs/registries/encoding.md:130]:
```suggestion
<table data-header-hidden><thead><tr><th width="164">Name</th><th>Value</th></tr></thead><tbody><tr><td>Signature</td><td><pre class="language-go"><code class="lang-go">ToRawJSON(v any) (string, error)
```

**Review by ccoVeille — COMMENTED** [review]
@42atomys what do you think?

**Comment by 42atomys** [comment]
@ccoVeille that correct ! But we cannot apply suggestion when the pull request are merged, I will apply in a new pr

**Pull request merged**